waku/v2/node/storage/message/waku_message_store.nim

```diff
@@ -12,22 +13,49 @@ import
 
 export sqlite
 
+logScope:
+  topics = "wakuMessageStore"
+
 const TABLE_TITLE = "Message"
+const maxStoreOverflow = 1.3
 
 # The code in this file is an adaptation of the Sqlite KV Store found in nim-eth.
 # https://github.com/status-im/nim-eth/blob/master/eth/db/kvstore_sqlite3.nim
-#
-# Most of it is a direct copy, the only unique functions being `get` and `put`.
 
 type
+  # WakuMessageStore implements auto deletion as follows:
+  # The sqlite DB will store up to `storeCapacity` * `maxStoreOverflow` messages.
+  # If more messages are stored, the messages are sorted by `receiverTimestamp` and the oldest ones are deleted,
+  # bringing the total number of stored messages back to `storeCapacity*maxStoreOverflow/2`. The rationale for batch deleting is efficiency.
+  # We keep half of the overflow window because we delete the oldest messages with respect to `receiverTimestamp` instead of `senderTimestamp`.
```

Comment:
```suggestion
  # We keep half of the db capacity because we delete the oldest messages with respect to `receiverTimestamp` instead of `senderTimestamp`.
```

Reply:
We actually keep capactiy + half of the overflow window. See comment above.
I will write this more clearly.